src/runtime_src/core/common/query_requests.h

```diff
@@ -29,6 +29,8 @@
 #include <boost/any.hpp>
 #include <boost/format.hpp>
 
+struct debug_ip_data;
```

Comment:
This doesn't look right. We have custom struct return types for many of these queries which are not declared as global variables

Reply:
This is forward declaration. I added it to avoid including the file which declares "debug_ip_data" datatype.

Reply:
Comment as forward declaration.   This is a pretty common header, which arguably could be split into multiple ones that are more domain specific, but until so, the includes should be at a minimum and types should be forward declared when that is sufficient.